pyproject.toml

```diff
@@ -0,0 +1,71 @@
+[project]
+name = "payu"
+authors = [
+    { name = "Marshall Ward", email = "python@marshallward.org" }
+]
+maintainers = [
+    { name = "ACCESS-NRI", email = "access.nri@anu.edu.au" }
+]
+description = "A climate model workflow manager for supercomputing environments"
+readme = "README.rst"
+license = { text = "Apache-2.0" }
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3",
+    "Topic :: Utilities",
+]
+keywords = ["climate model", "workflow"]
+dynamic = ["version"]
+# The dependencies here are also used by the conda build and so must follow
+# conda package match specifications, see here:
+# https://docs.conda.io/projects/conda-build/en/stable/resources/package-spec.html#package-match-specifications
+dependencies = [
+    "f90nml >=0.16",
+    "yamanifest >=0.3.4",
+    "PyYAML",
+    "requests",
+    "python-dateutil",
+    "tenacity >=8.0.0",
+    "cftime",
+    "GitPython >=3.1.40",
+    "ruamel.yaml >=0.18.5"
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest",
+    "pylint",
+    "Sphinx"
+]
+mitgcm = ["mnctools>=0.2"]
+
+[project.scripts]
+payu = "payu.cli:parse"
+payu-run = "payu.subcommands.run_cmd:runscript"
+payu-collate = "payu.subcommands.collate_cmd:runscript"
+payu-profile = "payu.subcommands.profile_cmd:runscript"
+payu-sync = "payu.subcommands.sync_cmd:runscript"
+payu-branch = "payu.subcommands.branch_cmd:runscript"
+payu-clone = "payu.subcommands.clone_cmd:runscript"
+payu-checkout = "payu.subcommands.checkout_cmd:runscript"
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools >= 61.0.0",
+  "versioneer[toml]"
+]
+
+[tool.setuptools.packages.find]
+include = ["payu*"]
+namespaces = false
+
+[tool.versioneer]
+VCS = "git"
+style = "pep440"
+versionfile_source = "payu/_version.py"
+versionfile_build = "payu/_version.py"
+tag_prefix = "v"
```

Comment:
The "v" doesn't get used in the version name. E.g. the tag `"v0.0.1" gives the version "`0.0.1`". So this still works with conda. I find this handy because it lets you create tags that don't change the version (so long as they don't start with "v"). Happy to change, but just wanted to make this clear

Reply:
So versioneer will work with the current tagging scheme (no `v` in the front)? It doesn't *require* the tags to start with `v`?

Reply:
It requires the `v`, so if we were to keep `tag_prefix` we would have to start tagging as `v?.?.?`

Reply:
I've removed the `tag_prefix`